### PR TITLE
Add hub admin FAQ pages

### PR DESCRIPTION
**Pull request by embarnard**
This PR adds admin access-able pages for creating, updating and destroying FaqCategories and FaqItems in the hub. You can find a link under "admin tools > frequently asked questions" or going directly to [/hub/faq](https://gyr-review-app-3602-4717051c0e8c.herokuapp.com/en/hub/faq).

### CODE REVIEW

File: app/controllers/hub/faq_items_controller.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        flash_message = "Successfully created '#{@faq_item.question_en}'"
+        render :show, notice: flash_message
+      else
+        flash_message = "Unable to create '#{@faq_item.question_en}', check validations"
+        render :new, error: flash_message
+      end
+    end
+
+    def edit
+      @position_options = FaqCategory.find(params[:faq_category_id]).faq_items.pluck(:position)
```
Comment (embarnard) [inline app/controllers/hub/faq_items_controller.rb:30]: add before action for @faq_category = FaqCategory.find(params[:faq_category_id])

File: app/views/hub/faq_items/_header.html.erb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<h1 class="h2"><%= link_to("FAQ Categories", hub_faq_categories_path) + " < " + link_to(item.faq_category.slug, hub_faq_category_path(id: item.faq_category)) + " < #{item.slug} (##{item.id})" %></h1>
+<div class="grid">
```
Comment (jalcine-cfa) [inline app/views/hub/faq_items/_header.html.erb:2]: non-blocking nit: make this check if `item` is provided and if not, render like show it's shown in `new.html.erb` (without the `item` info + deletion)

File: spec/controllers/hub/faq_categories_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+        faq_category.reload
+        expect(faq_category.versions.last.event).to eq "update"
+        expect(faq_category.versions.last.whodunnit).to eq user.id.to_s
+        expect(faq_category.versions.last.item_id).to eq faq_category.id
+        expect(faq_category.versions.last.item_type).to eq "FaqCategory"
+      end
+
+      context "slug is empty" do
+        let!(:slug) { "" }
+        it "updates the slug to the parameterized version of the name" do
```
Comment (jalcine-cfa) [inline spec/controllers/hub/faq_categories_controller_spec.rb:90]: non-blocking nit: should we check if the status code of the page is a 200 so it's not throwing an error for submitting an empty slug?

File: spec/controllers/hub/faq_items_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+  end
+
+  describe "#update" do
+    let(:slug) { "how_moon" }
+    let(:params) do
+      {
+        faq_category_id: faq_category.id,
+        id: faq_item.id,
+        faq_item: {
+          faq_category_id: faq_category.id,
```
Comment (jalcine-cfa) [inline spec/controllers/hub/faq_items_controller_spec.rb:37]: Do we have to resubmit this ID for updating?

File: spec/controllers/hub/faq_items_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+        expect(faq_item.versions.last.event).to eq "update"
+        expect(faq_item.versions.last.whodunnit).to eq user.id.to_s
+        expect(faq_item.versions.last.item_id).to eq faq_item.id
+        expect(faq_item.versions.last.item_type).to eq "FaqItem"
+      end
+
+      context "slug is empty" do
+        let!(:slug) { "" }
+        it "updates the slug to the parameterized version of the name" do
+          put :update, params: params
```
Comment (jalcine-cfa) [inline spec/controllers/hub/faq_items_controller_spec.rb:86]: Same here with checking of the status code to make sure it's a 200 (and no validations could have failed)

File: spec/controllers/hub/faq_items_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+    end
+  end
+
+  describe "#create" do
+    let(:params) do
+      {
+        faq_category_id: faq_category.id,
+        id: faq_item.id,
+        faq_item: {
+          faq_category_id: faq_category.id,
```
Comment (jalcine-cfa) [inline spec/controllers/hub/faq_items_controller_spec.rb:114]: Similar check to see if we need to resubmit this field (might be useful if we allow them to move items between categories, for example)

**Review by jalcine-cfa — COMMENTED** [review]
a few questions but LGTM!

**Pull request merged**